yarn-project/simulator/src/avm/opcodes/instruction.ts

```diff
@@ -14,14 +15,32 @@ type InstructionConstructor = {
  * It's most important aspects are execute and (de)serialize.
  */
 export abstract class Instruction {
+  /**
+   * Consumes gas and executes the instruction.
+   * This is the main entry point for the instruction.
+   * @param context - The AvmContext in which the instruction executes.
+   */
+  public run(context: AvmContext): Promise<void> {
```

Comment:
I'm torn about this! But I guess it's ok and it gives us a place to put the PC changes as well.

Reply:
Me too. I wanted an "external execute" that did the orchestration and an "internal execute" that had the logic, and let each instruction override either. Problem is I couldn't come up with good names.